src/main/java/edu/wisc/my/personalizedredirection/service/IRedirectionService.java

```diff
@@ -4,9 +4,9 @@
 
 import edu.wisc.my.personalizedredirection.controller.PersonalizedRedirectionController;
 import edu.wisc.my.personalizedredirection.dao.UrlDataSource;
-import edu.wisc.my.personalizedredirection.exception.PersonalizedRedirectionException;
+
 
 public interface IRedirectionService {
 
-    String getUrl(HttpServletRequest request, UrlDataSource dataSource) throws PersonalizedRedirectionException;
+    String getUrl(HttpServletRequest request, UrlDataSource dataSource) throws RuntimeException;
```

Comment:
I wouldn't declare `RuntimeException`. If I was going to declare it, I'd do it in a JavaDoc comment articulating the semantics of how this interface uses `RuntimeException`s in a Springy way.